plugins/org.eclipse.glsp.server/src/org/eclipse/glsp/server/features/core/model/ModelSubmissionHandler.java

```diff
@@ -87,15 +89,44 @@
     * </p>
     *
     * @param modelState The model state to submit.
+    * @param causedBy   The optional action/operation that caused the model update.
     * @return A list of actions to be processed in order to submit the model.
     */
-   public List<Action> submitModelDirectly(final GModelState modelState) {
+   public List<Action> submitModelDirectly(final GModelState modelState, final Action causedBy) {
       GModelRoot gModel = modelState.getRoot();
-      Action action = gModel.getRevision() == 0 ? new SetModelAction(gModel) : new UpdateModelAction(gModel);
+      DiagramConfiguration diagramConfiguration = diagramConfigurationRegistry.get(modelState);
+      if (diagramConfiguration.getLayoutKind() == ServerLayoutKind.AUTOMATIC) {
+         layoutEngine.layout(modelState);
+      }
+      Action modelAction = gModel.getRevision() == 0 ? new SetModelAction(gModel) : new UpdateModelAction(gModel);
```

Comment:
I already had cases where I had to override this method only to change the `animate` flag in this method. Would it be possible to extract the `new UpdateModelAction(gModel)` into a separate method or provide a method that returns the animate boolean that is then used in the creation of this action? I know it is not strictly part of this change but with this change the method just became more complicated and overriding more cumbersome ;-)

Reply:
I'm wondering if this is something that we could store in the `DiagramConfiguration` (similar to the needsClientLayout flag).  This way the user doesn't have to subclass the `ModelSubmissionHandler` and can configure it directly in the DiagramConfiguration.